src/date/index.ts

```diff
@@ -27,3 +31,344 @@ export function getMonthAbbrByIndex(monthNumber: number): string {
+ * only new Date()
+ * Not doing with dayjs as no specific methods for this use case are there.
+ *
+ * @param {Date} date - Date as only new Date()
+ * @param {number} date - number of days we want to go back to
+ *
+ * @example
+ * ```
+ * getPreviousDayDate(new Date());  // Output will be 'Thu Feb 03 2022 03:28:49 GMT+0530 (India Standard Time)'
+ * ```
+ */
+export function getPreviousDayDate(date: Date, daysCount: number = 1) {
+  try {
+    const newDate = new Date(date);
+
+    const newFinalDate = new Date(newDate.getTime() - (1000 * 60 * 60 * 24 * daysCount));
+
+    return newFinalDate;
+
+  } catch (error) {
+    console.error('Error in getPreviousDayDate: ', error);
+  }
+}
+
+
+/**
+ * This function converts DD/MM/YYYY to YYYY-MM-DD.
+ * JavaScript new Date() method accept date in YYYY-MM-DD or YYYY/MM/DD
+ * format that's why we needed this method
+ * This function is needed because dayjs does not work with 'DD/MM/YYYY' type of inputs
+ *
+ * @param {string} date - Input date string in DD/MM/YYYY or DD-MM-YYYY format
+ *
+ * @example
+ * ```
+ * convertStrToValidDateFormat('22/11/2021');  - // Output will be '2021-11-22'
+ * convertStrToValidDateFormat('03/25/2015');  - // Output will be '2015-25-03'
+ * convertStrToValidDateFormat('03-25-2015');  - // Output will be '2015-25-03'
+ * ```
+ */
+export function convertStrToValidDateFormat(date: string) {
+  try {
+    const str = date.replace(/[^0-9]/g, '');
+
+    if (str.length !== 8) {
+      throw new Error('Error in converting date\'s format');
+    }
+
+    const result = str.slice(4, 8) + '-' + str.slice(2, 4) + '-' + str.slice(0, 2);
+
+    return result;
+
+  } catch (err) {
+    console.error('Error in converting date\'s format', err);
+    return '';
+  }
+}
+
+
+/**
+ * This function checks if input date is valid or not.
+ *
+ * @param {string} date - Input date in YYYY-MM-DD format
+ * eg. 2021-11-22 (22nd November 2021)
+ * @param {string} delimiter - Provide delimiter in case date uses
+ * other delimiter than '-'.
+ *  * eg. Pass '/' in case date is in format of YYYY/MM/DD
+ *
+ * @example
+ * ```
+ * isValidDate();  // Output is false
+ * isValidDate('1993-10-10');  // Output is true
+ * isValidDate('1993/10/10');  // Output is false
+ * isValidDate('1993/10/10', '/');  // Output is true
+ * isValidDate('10/10/1993', '/');  // Output is false
+ * ```
+ */
+export function isValidDate(dateStr: string, delimiter: string = '-') {
+  try {
+    if (dateStr) {
+      const bits = dateStr.split(delimiter);
+      const date = new Date(bits[ 0 ], bits[ 1 ] - 1, bits[ 2 ]);
+
+      return date && (date.getMonth() + 1) === Number(bits[ 1 ]);
+
+    } else {
+      return false;
+    }
+
+  } catch (err) {
+    console.error('Error in validating date', err);
+    return false;
+  }
+}
+
+
+/**
+ * This function returns an array of objects which specify financial years from last supported year to offset
```

Comment:
what do u mean by last supported year, what this method does is still not clear after reading the comments

Reply:
updated comments